docs/api.md

```diff
@@ -0,0 +1,48 @@
+# Cloud Hypervisor API
+
+The Cloud Hypervisor API is made of 2 distinct interfaces:
+
+1. **The external API**. This is the user facing API. Users and operators can 
+   control and manage Cloud Hypervisor through either a REST API or a Command
+   Line Interface (CLI).
+
+2. **The internal API**, based on [Rust's Multi-Producer, Single-Consumer (MSPC)](https://doc.rust-lang.org/std/sync/mpsc/) module.
```

Comment:
MPSC:)

Reply:
I feel watched :-) Thanks @jiangliu, fixed now.

Reply:
We are redesign the API interface too, why you switch from restful API to non-restful API?

Reply:
The external API is still a REST API. We haven't changed anything, we're only documenting the external API and its relationship with the internal one.